src/Pkg3.jl

```diff
@@ -35,14 +44,44 @@ function _find_in_path(name::String, wd::Union{Void,String})
         path = joinpath(wd, name)
         Base.isfile_casesensitive(path) && return path
     end
+
     info = Pkg3.Operations.package_env_info(base, verb = "use")
-    info == nothing && return nothing
-    haskey(info, "uuid") || return nothing
-    haskey(info, "hash-sha1") || return nothing
+    info == nothing && @goto find_global
+    haskey(info, "uuid") || @goto find_global
+    haskey(info, "hash-sha1") || @goto find_global
     uuid = Base.Random.UUID(info["uuid"])
     hash = Pkg3.Types.SHA1(info["hash-sha1"])
     path = Pkg3.Operations.find_installed(uuid, hash)
-    ispath(path) ? joinpath(path, "src", name) : nothing
+    ispath(path) && return joinpath(path, "src", name)
+
+    # If we still haven't found the file, look if the package exists in the registry
+    # and query the user (if we are interactive) to install it.
+    @label find_global
+    if isinteractive()
+        ec = Types.EnvCache()
```

Comment:
I've been calling this variable `env` typically. Maybe change it for consistency?